com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs

```diff
@@ -152,11 +153,12 @@ SensorComponent[] sensorComponents
                 if (sensor.GetObservationShape().Length == 2)
                 {
                     if (!tensorsNames.Contains(
-                        TensorNames.ObservationPlaceholderPrefix + sensorIndex))
+                        TensorNames.ObservationPlaceholderPrefix + varLenIndex))
```

Comment:
What's going on here? I thought the variable length inputs were labeled by the sensor index? 
(maybe if I keep reading it will answer my question)

Reply:
This is rather tricky. The inputs are based on sensor indexes yes, but this is sensor components.
I think this whole code is a ticking bomb, because we only have access to the sensor components, not the sensors (because the agent was not initialized) and we assume their order is going to be consistent. This is probably not the case going forward since we now have both visual and var len obs. I need to refactor this whole thing but I have no idea how I can get the list of ISensors BEFORE the Agent is initialized...

Reply:
Arg, yeah, the sensors are sorted by name, but we don't have any restrictions on the sensor components...

Reply:
Trying to find a workaround, will make a separate PR to this one if I find something.